webapp/src/components/NewsList.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <div>
+    <news-item
+        v-for="newsItem in sortedItems"
+        v-bind:title="newsItem.title"
+        v-bind:votes="newsItem.votes"
+        v-bind:key="newsItem.title"
+        v-on:upvoteItem="upvoteNews"
+        v-on:downvoteItem="downvoteNews"
+        v-on:removeNews="removeNews"
+    ></news-item>
+    <div class="form-div">
+      <form v-on:submit.prevent class="form">
+        <input v-model="newItemTitle" placeholder="News Item Title here">
+        <button v-on:click="addItem">Insert item</button>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script>
+
+import NewsItem from "@/components/NewsItem";
+
+export default {
+  name: "NewsList",
+  components: {NewsItem},
+  data: function () {
+    return {
+      newsItems: [
+        {title: "Title_1", votes: 0},
+        {title: "Title_2", votes: 0}
+      ],
+      newItemTitle: "",
+    }
+  },
+  methods: {
+    upvoteNews(title) {
+      let toBeUpvoted = this.newsItems.find(item => item.title === title)
+      toBeUpvoted.votes += 1
```

Comment:
Put this logic into the child component and then `this.$emit('updateItem', item)`. Here in the parent component, you should only replace the item:
```js
this.newsItems = this.newsItems.map(i => i.someUniqueId  === item.someUniqueId ? item : i)
```